### PR TITLE
#846: Define and export strict Dropdown's Props types (closes #846)

### LINKED ISSUE #846: Define and export strict Dropdown's Props types
## description

Dropdown loosely pass down to `Select` every other undeclared props
I think it's safer to handle it as a "strict props" component

## specs
- use `strict: true` in `@props`
- add missing props types in `Props` used in several projects (just to make it as least breaking as possible)

### PR DESCRIPTION
**Pull request by francescogior**
Closes #846

### CODE REVIEW

File: src/dropdown/Dropdown.js [hunk 1]
```diff
@@ -54,13 +60,20 @@ export const Props = {
...
  * @param simpleValue - if true, selected values will be passed to onChange as comma-separated string of values (eg "1,2,3") instead of array of objects
+ * @param optionRenderer - the function that can be used to override the default renderer of options
+ * @param valueRenderer - the function that can be used to override the default renderer of the selected value
  * @param menuRenderer - the function that can be used to override the default drop-down list of options
  * @param groupByKey - the field name to group by
  * @param optionGroupRenderer - the function that gets used to render the content of an option group
  * @param menuPosition - whether the menu should be rendered on top or bottom when it's open
+ * @param placeholder - placeholder shown when no value is selected
+ * @param noResultsText - if searchable, message shown in the menu when no results are found
+ * @param allowCreate - whether it should be possible to create new options
```
Comment (ascariandrea) [inline src/dropdown/Dropdown.js]: There's no `allowCreate` prop defined in `@props`

File: src/dropdown/Dropdown.js [hunk 2]
```diff
@@ -54,13 +60,20 @@ export const Props = {
...
+ * @param addLabelText - if allowCrete is true, message shown to hint the user to press Enter to create a new option
```
Comment (ascariandrea) [inline src/dropdown/Dropdown.js]: s/allowCrete/allowCreate/

Comment (francescogior) [inline src/dropdown/Dropdown.js]: I read 'allowCreAte', what do you mean?

Comment (francescogior) [inline src/dropdown/Dropdown.js]: Oh got it :)

File: src/dropdown/Dropdown.js [hunk 3]
```diff
@@ -54,13 +61,20 @@ export const Props = {
...
  * @param menuPosition - whether the menu should be rendered on top or bottom when it's open
+ * @param placeholder - placeholder shown when no value is selected
+ * @param noResultsText - if searchable, message shown in the menu when no results are found
+ * @param allowCreate - whether it should be possible to create new options
+ * @param addLabelText - if allowCreate is true, message shown to hint the user to press Enter to create a new option
+ * @param delimiter - if multi is true, string used to separate selected values
  */
 @skinnable()
-@props(Props, { strict: false })
+@props(Props, { strict: true })
```
Comment (FrancescoCioria) [inline src/dropdown/Dropdown.js]: (`strict: true` is the default)

**Pull request merged**